src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.Scanner;
+
+public class Parser {
+    public static void parse(String input, Scanner sc, TaskList tasks, Storage storage) {
+        while (!input.equalsIgnoreCase("bye")) {
+            Ui.showLine();
+            int id;
+            switch(input) {
+                case "list" :
+                    tasks.list();
+                    break;
+                case "todo" :
+                    sc.reset();
+                    try {
+                        input = sc.nextLine().strip();
+                        if (input.isEmpty()) throw new DukeException("Do what, sir?");
```

Comment:
Shouldn't there be Egyptian braces for your if statements? 🤔
I noticed this for your if statements in this class.